pkg/core/subscription.go

```diff
@@ -0,0 +1,6 @@
+package core
+
+type Subscription interface {
```

Comment:
A lot of the size of this PR comes from putting the Subscription behind an interface, which I did so that we could send an error to it in the tests. Felt like that interface belonged here, but that led to import cycles with the `client` package, which meant further shuffling